### DISCUSSION TITLE
Combobox onChange with Combobox.Input's onBlur execution

### DISCUSSION THREAD

**Discussion by glRajkumar2** [tailwindlabs/headlessui#1748] [Help]
I am using onBlur event in the Combobox.Input component. If i use keyboard to select a option in the list. there is no problem. The item first selected then onBlur event called. But if i try to select a option by cursor, then onBlur is called first and old value is used instead of the option value.

```
<Combobox value={value} onChange={setValue}>
     <Combobox.Input 
         onChange={updateVal} 
         onBlur={onBlur}
      />
      <Combobox.Options>
        {.....}
      </Combobox.Options>
</Combobox>
```

**Comment by RobinMalfait** [comment]
Hey! Thank you for your question!
Much appreciated! 🙏

What are you doing in the `onBlur` exactly?

If you use your mouse to click an option, then you will blur the `input`, click the `option`. There is nothing we can do about that it's just how the browser works.

**Reply by glRajkumar2** [reply]
For my requirement, user can enter or go to next section without enter from the input. so validating user input on onBlur event.

**Reply by glRajkumar2** [reply]
Also another point, onChange function is not called if onBlur executed. is there any way to use onClick on Combobox.Option?
